tests/integration/test_workflow_simplifications.py

```
@@ -0,0 +1,652 @@
+                key = loader.construct_object(key_node, deep=False)
+                if key in mapping:
+                    duplicates.append(key)
+                mapping[key] = loader.construct_object(value_node, deep=False)
+            return mapping
+
+        DuplicateKeySafeLoader.add_constructor(
+            yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, constructor_with_dup_check
+        )
+
+        try:
+            yaml.load(content, Loader=DuplicateKeySafeLoader)
```

Ruff (0.15.1)

[error] Probable use of unsafe loader `DuplicateKeySafeLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)